### ISSUE TITLE
Add possibility to change toolchain (CC, AR, AS, LINKER, DUMP) when building

### ISSUE THREAD

**Issue by roarbr** [Xilinx/embeddedsw#150]
We need a newer compiler than petalinux provides. We have built our own toolchains for aarch64-elf, microblaze-elf and aarch64-linux-gnu with the vendor-part of toolchain tuple set. We use these toolchains to build ATF, PMUFW and FSBL.

So our compiler is not called "aarch64-none-elf-gcc" (ATF, FSBL) or "mb-gcc" (PMUFW), but these compiler names are hard coded in Makefiles within the embeddedsw git repo.

In embeddedsw Makefiles the CC variable is set like this:
CC := "mb-gcc"

This is a hard setting that we cannot override from the outside as far as I know (?). But if the assignment was changed to this:
CC ?= "mb-gcc"

Then it will only be set to this variable if not already defined. 
By using "CC ?= "mb-gcc" we can override CC when building:
CC="microblaze-MyVendorTag-elf-gcc" make

And some places this type of code exists to call submake process:
$(MAKE) [....] "COMPILER=mb-gcc" ARCHIVER="mb-ar"

Should be changed to this to inherit top level toolchain settings:
$(MAKE) [....] "COMPILER=$(CC)" ARCHIVER="$(AR)"

Same for other toolchain variables like AS, AR, DUMP and LINKER.

We have patched some Makefiles for now to make this work, but ideally your Makefiles should be updated.

Would you consider changing Makefiles to the other assignment version (?=) and
also consider removing hard coded variables like COMPILER and ARCHIVER and use variables when calling submake processes?

**Comment by sivadur** [comment]
@roarbr I agree with your suggestion, Thanks for it. We will look into this.
